tests/test_models.py

```diff
@@ -608,39 +620,124 @@ def test_sweep_coalescence_same_seed(self):
 
-    @unittest.skip("Can't get sweep to be incomplete")
     def test_sweep_start_time_incomplete(self):
         # Short sweep that doesn't make complete coalescence.
         sweep_model = msprime.SweepGenicSelection(
             reference_size=0.25, position=0.5, start_frequency=0.69,
-            end_frequency=0.7, alpha=1e-5, dt=1)
+            end_frequency=0.7, alpha=8, dt=0.001)
         t_start = 0.1
         ts = msprime.simulate(
             10, Ne=0.25,
             recombination_rate=2,
             demographic_events=[
                 msprime.SimulationModelChange(t_start, sweep_model)],
-            num_labels=2, random_seed=2)
+            random_seed=2)
         self.assertTrue(any(tree.num_roots > 1 for tree in ts.trees()))
+
+    def test_sweep_model_change_time_complete(self):
+        # Short sweep that doesn't coalesce followed
+        # by Hudson phase to finish up coalescent
+        sweep_model = msprime.SweepGenicSelection(
+            reference_size=0.25, position=0.5, start_frequency=0.69,
+            end_frequency=0.7, alpha=1e-5, dt=1)
+        ts = msprime.simulate(
+            10, Ne=0.25,
+            recombination_rate=2,
+            demographic_events=[
+                msprime.SimulationModelChange(0, sweep_model),
+                msprime.SimulationModelChange(None, "hudson")],
+            random_seed=2)
+        self.assertTrue(all(tree.num_roots == 1 for tree in ts.trees()))
+        # Make sure that the Hudson phase did something.
+        ts = msprime.simulate(
+            10, Ne=0.25,
+            recombination_rate=2,
+            demographic_events=[
+                msprime.SimulationModelChange(0, sweep_model)],
+            random_seed=2)
+        self.assertTrue(any(tree.num_roots > 1 for tree in ts.trees()))
+
+    def test_many_sweeps(self):
+        sweep_models = [
+            msprime.SweepGenicSelection(
+                reference_size=0.25, position=j, start_frequency=0.69,
+                end_frequency=0.7, alpha=1e-5, dt=0.1)
+            for j in range(10)]
+        ts = msprime.simulate(
+            10, Ne=0.25, length=10,
+            recombination_rate=0.2,
+            demographic_events=[
+                msprime.SimulationModelChange(0.01, sweep_models[0])] + [
+                    msprime.SimulationModelChange(None, model)
+                    for model in sweep_models] + [
+                msprime.SimulationModelChange()],
+            random_seed=2)
+        self.assertTrue(all(tree.num_roots == 1 for tree in ts.trees()))
+
+    def test_many_sweeps_regular_times_model_change(self):
+        demographic_events = []
+        for j in range(10):
+            sweep_model = msprime.SweepGenicSelection(
+                position=j, start_frequency=0.69, end_frequency=0.7,
+                alpha=1000, dt=0.0125)
+            # Start the sweep after 0.01 generations of Hudson
+            demographic_events.append(msprime.SimulationModelChange(
```

Comment:
So, this is the proposal for doing lots of sweeps. The ``event.time`` field is a callable, which returns the time at which the sweep should start, given the current simulation time at the start of the previous model as a parameter. (This is slightly confusing, but general, I think). This way we don't have to make decisions before the simulation is run and can have randomness in there too.

Reply:
> given the current simulation time at the start of the previous model as a parameter. (This is slightly confusing, but general, I think). 

this is a bit confusing. why not just hand it the current time of the simulation at exit from the previous model?

Reply:
> this is a bit confusing. why not just hand it the current time of the simulation at exit from the previous model?

Because we're generating the time that the previous model finishes. The way to think of it is this: we're about to run a Hudson phase at time t, so what time should this Hudson phase finish so we can start a sweep?

Reply:
I like it, but it needs to be a bit better explained in the docs

Reply:
OK, will open an issue to track this doc task.